react-intl/index.d.ts

```diff
@@ -12,7 +12,7 @@ declare namespace ReactIntl {
         pluralRuleFunction?: (n: number, ord: boolean) => string;
     }
 
-    function injectIntl<TOriginalProps, TOwnProps>(component: React.ComponentClass<TOriginalProps> | React.StatelessComponent<TOriginalProps>): React.ComponentClass<TOwnProps>;
+    function injectIntl<P> (component: React.ComponentClass<P>): React.ComponentClass<P>;
```

Comment:
change this line to:
```TypeScript
function injectIntl<T extends React.ComponentClass<InjectedIntlProps> | React.StatelessComponent<InjectedIntlProps> >(component: T): T;
```
.. so it will work for stateless components as well